src/align/anchors.jl

```diff
@@ -255,17 +255,54 @@ immutable AlignedSequence{S}
     aln::Alignment
 end
 
-
-function AlignedSequence{S}(seq::S, aln::Alignment)
-    return AlignedSequence{S}(seq, aln)
-end
-
-
 function AlignedSequence{S}(seq::S, anchors::Vector{AlignmentAnchor},
                             check::Bool=true)
     return AlignedSequence(seq, Alignment(anchors, check))
 end
 
+function AlignedSequence(seq::Sequence, ref::Sequence)
+    return AlignedSequence(seq, 1, ref, 1)
+end
+
+function AlignedSequence(seq::Sequence, seqpos::Integer,
+                         ref::Sequence, refpos::Integer)
+    if length(seq) != length(ref)
+        throw(ArgumentError("two sequences must be the same length"))
+    end
+    seqpos -= 1
+    refpos -= 1
+    op = OP_START
+    newseq = similar(seq, 0)  # sequence without gap symbols
+    anchors = AlignmentAnchor[]
+    for (x, y) in zip(seq, ref)
+        if x == gap(eltype(seq)) && y == gap(eltype(ref))
+            throw(ArgumentError("two sequences must not have gaps at the same position"))
```

Comment:
Exactly. This constructor is intended for pairwise alignment, but, yes, we need to care about MSA so that we can create an `AlignedSequence` for each aligned sequence even when gaps occur at the same positions.